server/teaching-element/te.controller.js

```diff
@@ -4,18 +4,16 @@ const { Activity, TeachingElement } = require('../shared/database');
 const { createError } = require('../shared/error/helpers');
 const { NOT_FOUND } = require('http-status-codes');
 const { Op } = require('sequelize');
-const { resolveStatics } = require('../shared/storage/helpers');
 const pick = require('lodash/pick');
 
 function list({ course, query, opts }, res) {
-  if (query.activityId || query.parentId) {
-    const { activityId, parentId } = query;
-    const where = { [Op.or]: [] };
-    if (activityId) where[Op.or].push({ id: parseInt(activityId, 10) });
-    if (parentId) where[Op.or].push({ parentId: parseInt(parentId, 10) });
+  if (!query.detached) opts.where = { detached: false };
+  if (query.ids) {
+    const ids = query.ids.map(id => Number(id));
+    const cond = { [Op.in]: ids };
```

Comment:
We always had it and it was because of Exam and the fact that they have sub-containers. We need to change the entire way we load teaching elements to remove it.